ivy/functional/backends/jax/random.py

```diff
@@ -57,7 +57,13 @@ def random_uniform(
     else:
         RNG_, rng_input = jax.random.split(_getRNG())
         _setRNG(RNG_)
-
+    if dtype in [jnp.int32, jnp.int64]:
+        return to_device(
+            jax.random.uniform(
+                rng_input, shape, minval=low, maxval=high, dtype=jnp.float32
+            ),
+            device,
+        ).astype(jnp.int64)
```

Comment:
As per the [documentation](https://lets-unify.ai/docs/ivy/docs/functional/ivy/ivy.functional.ivy.random.html#ivy.functional.ivy.random.random_uniform) of `ivy.random_uniform()` the output tensor is of `dtype`, doesn't that makes it better to remove the if condition entirely and instead always cast to `dtype`. 

It seemed like a bug that we don't do that casting

Reply:
i will fix it